lib/api/api.go

```diff
@@ -874,7 +870,13 @@ func (s *service) getDebugFile(w http.ResponseWriter, r *http.Request) {
 
 func (s *service) postSystemRestart(w http.ResponseWriter, r *http.Request) {
 	s.flushResponse(`{"ok": "restarting"}`, w)
-	go s.contr.Restart()
+
+	go func() {
```

Comment:
`go`ing these feels kind of dirty. How about making the channel one-buffered and doing these sends with a select, maybe returning an error if we take the default clause (`"shutdown in progress"`)?